crates/storage/provider/src/providers/database/provider.rs

```diff
@@ -1235,6 +1235,66 @@ impl<TX: DbTx> BlockNumReader for DatabaseProvider<TX> {
     }
 }
 
+impl<Tx: DbTx> DatabaseProvider<Tx> {
+    fn process_block_range<F, R>(
+        &self,
+        range: RangeInclusive<BlockNumber>,
+        mut process_tx: F,
+    ) -> ProviderResult<Vec<R>>
+    where
+        F: FnMut(Range<TxNumber>, Header, Vec<Header>, Option<Withdrawals>) -> ProviderResult<R>,
+    {
+        if range.is_empty() {
+            return Ok(Vec::new())
+        }
+
+        let len = range.end().saturating_sub(*range.start()) as usize;
+        let mut blocks = Vec::with_capacity(len);
+
+        let mut headers_cursor = self.tx.cursor_read::<tables::Headers>()?;
```

Comment:
there was recently a PR that fixed an issue in this snippet of code

https://github.com/paradigmxyz/reth/pull/7429

would need the same fix

Reply:
Done.